src/languageservice/services/yamlCodeLens.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { CodeLens, Range } from 'vscode-languageserver-types';
+import { YamlCommands } from '../../commands';
+import { yamlDocumentsCache } from '../parser/yaml-documents';
+import { YAMLSchemaService } from './yamlSchemaService';
```

Comment:
I wonder if we should start making a standard for exported classes/interfaces that start with yaml. Right now we have 3 different styles from lines 8-10:
```
Yaml
yaml
YAML
```

Reply:
If we treat it as a word that is pronounced then I think it should be `yaml` and `Yaml` depending if the first letter is capitalized. 
I know that I do not say Y A M L so lowercase works for me.

Reply:
Yes we need naming conventions, not only for classes/interfaces but for TS file names also.
But I don't think that it should be a part of this PR. In this PR I add `yamlDocumentsCache` which is const instance of `YamlDocuments` class. `YAMLSchemaService` was created a way before, I think it name collide with `JSONSchemaService` which it extends.

Reply:
Yeah, I don't think we need to address it in this PR, we can address it in a different one. It was just something I noticed when looking at the imports